data/time_series.go

```diff
@@ -82,6 +95,58 @@ func (f *Frame) TimeSeriesSchema() (tsSchema TimeSeriesSchema) {
 	return
 }
 
+// valueToType gets a float64 value and converts it to the specific field type
+// this is useful is fill missing is enabled and fill missing mode is FillMissingValue
+// for converrting the fill missing value (float64) to the field type
+func valueToType(val float64, ftype FieldType) interface{} {
+	switch ftype {
+	case FieldTypeInt8, FieldTypeNullableInt8:
+		return int8(val)
+	case FieldTypeInt16, FieldTypeNullableInt16:
+		return int16(val)
+	case FieldTypeInt32, FieldTypeNullableInt32:
+		return int32(val)
+	case FieldTypeInt64, FieldTypeNullableInt64:
+		return int64(val)
+	case FieldTypeUint8, FieldTypeNullableUint8:
+		return uint8(val)
+	case FieldTypeUint16, FieldTypeNullableUint16:
+		return uint16(val)
+	case FieldTypeUint32, FieldTypeNullableUint32:
+		return uint32(val)
+	case FieldTypeUint64, FieldTypeNullableUint64:
+		return uint64(val)
+	case FieldTypeFloat32, FieldTypeNullableFloat32:
+		return float32(val)
+	case FieldTypeFloat64, FieldTypeNullableFloat64:
+		return val
+	}
+	// if field type is FieldTypeString, FieldTypeNullableString, FieldTypeBool, FieldTypeNullableBool, FieldTypeTime, FieldTypeNullableTime
```

Comment:
I'm also wondering about whether an error should be returned if the field type isn't handled.

Reply:
FloatAt currently supports all these types, and just does things according to its commented rules.

It will return an error in case a parsefloat error. But even without that I still would have returned an Error, because presumably we will add types where a corresponding float representation can't make any sense.

`func (f *Field) FloatAt(idx int) (float64, error)`

So ideally I think we would replace this with SetFloatAt, which would be exported. 

But we can just add some comments if we want to save this for another day since it is not exported. If going that route, probably good to return an error in this case for now.